### ISSUE TITLE
Issue with determining machine architecture

### ISSUE THREAD

**Issue by Marcool04** [droserasprout/pacman-fix-permissions#10]
Hi there,
Just ran into an issue trying to use the script on archlinuxarm on a raspberrypi 4B+.
The problem is here: https://github.com/droserasprout/pacman-fix-permissions/blob/161db3fd2e2ec4d6db2899c89fc49a7922cb63a5/src/pacman_fix_permissions/__init__.py#L81

The problem is that `uname -m` doesn't give the correct architecture, from pacman's perspective, at least on my machine:

````
[root@hostname ~]# grep Architecture /etc/pacman.conf
Architecture = armv7h
[root@hostname ~]# uname -m
armv7l
````

And packages are indeed build as `pkgname-pkgver-armv7h` and not `-armv7l`, which causes the script to fail with:

````
[snip]
==> Parsing installed packages list
==> Collecting actual filesystem permissions and correct ones from packages
(1/942) ack 3.4.0-1
(2/942) acl 2.2.53-3
=> acl package is missing, downloading
Traceback (most recent call last):
  File "/usr/bin/pacman-fix-permissions", line 33, in <module>
    sys.exit(load_entry_point('pacman-fix-permissions==1.1.1', 'console_scripts', 'pacman-fix-permissions')())
  File "/usr/lib/python3.9/site-packages/pacman_fix_permissions/__init__.py", line 136, in __main__
    pkgtar = getTar(pkgs[i])
  File "/usr/lib/python3.9/site-packages/pacman_fix_permissions/__init__.py", line 89, in getTar
    raise Exception(
Exception: Can't open or download 'acl' package, check your internet connection
````

**Comment by Marcool04** [comment]
I've proposed a fix in pull request #11.

**Comment by droserasprout** [comment]
That's an interesting case. According to `man pacman.conf` current architecture is determined with `uname -m` command when using `Architecture = auto` (default for x86_64). Does it mean replacing `armv7h` with `auto` in your pacman.conf will break things or there's some ARM tricks I'm not aware of?

**Comment by Marcool04** [comment]
Oh… this is waaaay more complicated that I thought 🧐
It appears the nuance between h and l is not so much one of architecture, but of how binaries have been compiled (or will be, if we're looking at makepkg / pacman): h is for “hard float”, l is for “little endian”.
I'm just learning here, have read around a bit, and [this](https://stackoverflow.com/questions/29166619/differences-between-arm-versions-armv7-only) seems to summarize it nicely: 

> I would assume that it's indicating packages compiled for little-endian and hard-float ABI as appropriate - i.e. it's a software thing and only tangentially related to the hardware.
> 
> In other words, you don't actually have an "armv7l" processor - you have an ARMv7 processor which may well have a hardware FPU (and can run big-endian if you really wanted to), but you happen to be running a soft-float userspace that doesn't rely on one being present - just like running an i686 distribution doesn't imply you're not on an x86_64 machine. Different Linux distributions have different names for their various ports but some trivial poking around suggests this case might be openSUSE's convention.

Sorry to drag arm problems into the arch world (where I understand the standard is just to support x86_64). I need to get some stuff in order then I'll take a backup and see what happens when I switch `Architecture` to `auto`… 😰 It probably will fail to find packages that are in the cache (since they're called `-armv7h`, and then start building everything to packages ending in `-armv7l`, and work just fine, if I understand the situation ok.

Thanks for taking the time to look into it anyhow. Will post back soon.
Regards,
Mark.

**Comment by droserasprout** [comment]
Fixed in 1.1.2

**Comment by Marcool04** [comment]
Oh wow that was quick :) I still haven't had time to test the "auto" case on my machine. Looks like it'll work. Thanks for the quick attention to my issue.
